forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,391 @@
+    sync_status: SyncStatus,
+    /// Start time of the node
+    start_time: DateTime<Utc>,
+    /// Current network the node is running on
+    network: String,
+    /// Default wallet address selected.
+    default_wallet_address: Option<String>,
+}
+
+impl NodeStatusInfo {
+    fn set_default_wallet(&mut self, wallet_addr: Option<String>) {
+        self.default_wallet_address = wallet_addr;
+    }
+
+    fn set_network(&mut self, network: &str) {
+        self.network = network.to_string();
+    }
+
+    fn set_node_start_time(&mut self, start_time: DateTime<Utc>) {
+        self.start_time = start_time
+    }
+
+    fn new(
+        cur_duration: Duration,
+        tipsets: Vec<TipsetJson>,
+        chain_finality: usize,
+    ) -> anyhow::Result<NodeStatusInfo> {
+        let head = tipsets
+            .get(0)
+            .map(|ts| ts.0.clone())
+            .ok_or(anyhow::anyhow!("head tipset not found"))?;
+        let num_tipsets = tipsets.len().max(chain_finality);
+        let block_count: usize = tipsets.iter().map(|s| s.0.blocks().len()).sum();
+        let epoch = head.epoch();
+        let ts = head.min_timestamp();
+        let cur_duration_secs = cur_duration.as_secs();
+        let behind = if ts <= cur_duration_secs + 1 {
+            cur_duration_secs.saturating_sub(ts)
+        } else {
+            anyhow::bail!(
+                "System time should not be behind tipset timestamp, please sync the system clock."
+            );
+        };
+
+        let sync_status = if behind < EPOCH_DURATION_SECONDS as u64 * 3 / 2 {
+            // within 1.5 epochs
+            SyncStatus::Ok
+        } else if behind < EPOCH_DURATION_SECONDS as u64 * 5 {
+            // within 5 epochs
+            SyncStatus::Slow
+        } else {
+            SyncStatus::Behind
+        };
+
+        let base_fee = head.min_ticket_block().parent_base_fee().clone();
+
+        let health = (100 * block_count) as f64 / (num_tipsets * BLOCKS_PER_EPOCH as usize) as f64;
+
+        Ok(NodeStatusInfo {
+            behind: Duration::from_secs(behind),
+            health,
+            epoch,
+            base_fee,
+            sync_status,
+            start_time: Utc::now(),
+            network: String::from("unknown"),
+            default_wallet_address: None,
+        })
+    }
+
+    fn display(&self, color: &LoggingColor) -> NodeInfoOutput {
+        let NodeStatusInfo { health, .. } = self;
+
+        let use_color = color.coloring_enabled();
+        let uptime = (Utc::now() - self.start_time)
+            .to_std()
+            .expect("failed converting to std duration");
+        let fmt_uptime = fmt_duration(uptime);
+        let uptime = format!(
+            "{fmt_uptime} (Started at: {})",
+            self.start_time.with_timezone(&chrono::offset::Local)
+        )
+        .normal();
+
+        let chain_status = self.chain_status().blue();
+        let network = self.network.green();
+        let wallet_address = self
```

Comment:
Could be nice to add balance as well since we're here, like in lotus:

`fmt.Printf("      %s [%s]\n", addr.String(), types.FIL(balance).Short())`